caikit_nlp/modules/text_generation/peft_tgis_remote.py

```diff
@@ -40,24 +40,34 @@
 error = error_handler.get(log)
 
 
-@block(backend_type=TGISBackend.backend_type, base_module=PeftPromptTuning)
+@modules.module(backend_type=TGISBackend.backend_type, base_module=PeftPromptTuning)
 class PeftPromptTuningTGIS(ModuleBase):
     SUPPORTED_LOAD_BACKENDS = [TGISBackend.backend_type, backend_types.LOCAL]
     ## Module Interface ##
 
-    def __init__(self, base_model_name, prompt_cache_id, eos_token, verbalizer) -> None:
+    def __init__(
+            self,
+            base_model_name,
+            prompt_cache_id,
+            eos_token,
+            verbalizer,
+            enable_backend=True,
+            tgis_backend=None,
+        ) -> None:
         super().__init__()
         # Configure the internal client
-        self._client = module_backend_config.get_backend(
-            TGISBackend.backend_type
-        ).get_client(base_model_name)
+        if enable_backend:
+            # get_client will also launch a local TGIS process and get the model
+            # loaded when using the local tgis backend
+            self._client = tgis_backend.get_client(base_model_name)
+
         self.base_model_name = base_model_name
         self._prompt_cache_id = prompt_cache_id
         self.eos_token = eos_token
         self.verbalizer = verbalizer
 
     @classmethod
-    def load(cls, model_path: str) -> "PeftPromptTuningTGIS":
+    def load(cls, model_path: str, load_backend: BackendBase) -> "PeftPromptTuningTGIS":
```

Comment:
Given that this backend is TGIS specific, do we need additional validation here? I'm wondering what the advantage of parameterizing this are

Reply:
I see - but will those ever be instances that aren't `TGISBackend`? I am mostly wondering if we should throw a TypeError if a different implementation of the BackendBase is passed (except for testing via the stub stuff maybe)

Reply:
Likely not. I'll add error check